syllabus/curricula/plr/plr_wrapper.py

```diff
@@ -244,7 +280,10 @@ def sample(self, k: int = 1) -> Union[List, Any]:
         if self._should_use_startup_sampling():
             return self._startup_sample()
         else:
-            return [self._task_sampler.sample() for _ in range(k)]
+            if self._use_robust_plr:  # Check if Robust PLR is enabled
+                return [self._evaluator.sample_task() for _ in range(k)]  # Use evaluator to sample tasks
+            else:
+                return [self._task_sampler.sample() for _ in range(k)]  # Use task sampler
```

Comment:
If we're using robust PLR, we still want to use the task sampler as normal. The difference is that if the task sampler selects  a random task instead of a replay task, we want to evaluate it without returning it to the user, then sample a different task from replay for them to train on. The task sampler makes the replay vs. random decision here https://github.com/RyanNavillus/Syllabus/blob/55a0acaaf391b3cb2ed814ba4ad64751aa8dd5cd/syllabus/curricula/plr/task_sampler.py#L264

There are a few options for how to do the update. We can evaluate them immediately, and return a new task after we update the task scores, but that might be slow because we can't send a new task until the evaluation is done. Alternatively, we could collect the tasks into a queue, and then evaluate a bunch of them at once. I'm not sure which one is better right now, so I suggest you do whatever is easiest and then we can improve from there